nightfall-optimist/src/services/block-assembler.mjs

```diff
@@ -129,16 +130,19 @@ export async function conditionalMakeBlock(proposer) {
         // before sending Proposed block. If not wait until the proposer reconnects
         let tryCount = 0;
         while (!ws || ws.readyState !== WebSocket.OPEN) {
-          await new Promise(resolve => setTimeout(resolve, 3000)); // eslint-disable-line no-await-in-loop
+          await waitForTimeout(3000); // eslint-disable-line no-await-in-loop
 
-          logger.warn(`Websocket to proposer is closed.  Waiting for proposer to reconnect`);
+          logger.warn(`Websocket to proposer is closed. Waiting for proposer to reconnect`);
 
           increaseProposerWsClosed();
           if (tryCount++ > 100) {
             increaseProposerWsFailed();
-            throw new Error(`Websocket to proposer has failed`);
```

Comment:
Why don’t you throw error?

Reply:
Because this crashes the Optimist. I don't think this is the right behavior, no one would like to have its app crashing just because it wasn't possible to connect to a Proposer. There is no `try...catch` to catch this. It should retry until it gets a connection. Btw, I'll add an `logger.error` to warn about this.